### PR TITLE
Fix: Icon change position not centered & SVG icon is not visible in FE [ED-8316]

### PR DESCRIPTION
**Pull request by davseve**

### CODE REVIEW

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 1]
```diff
@@ -141,8 +146,7 @@
 					}
 
 					svg {
-						width: var( --n-tabs-icon-size );
-						height: var( --n-tabs-icon-size );
+						height: var( --n-tabs-svg-icon-size );
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:149]: Why has this been removed?

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 2]
```diff
@@ -47,10 +47,12 @@
 	--n-tabs-title-border-radius: initial;
 	--n-tabs-title-transition: 0.3s;
 
+  	--n-tabs-icon-align-items: center;
 	--n-tabs-icon-color: var( --e-global-color-secondary, var( --n-tabs-color-secondary-fallback ) );
 	--n-tabs-icon-color-hover: var( --n-tabs-title-color-hover );
 	--n-tabs-icon-color-active: #{$white};
 	--n-tabs-icon-size: initial;
+  	--n-tabs-svg-icon-size: 15px;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:55]: Why has this been added?

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 3]
```diff
@@ -785,7 +785,7 @@ protected function register_controls() {
 			],
 			'size_units' => [ 'px', 'em', 'rem' ],
 			'selectors' => [
-				'{{WRAPPER}}' => '--n-tabs-icon-size: {{SIZE}}{{UNIT}}',
+				'{{WRAPPER}}' => '--n-tabs-svg-icon-size: {{SIZE}}{{UNIT}}; --n-tabs-icon-size: {{SIZE}}{{UNIT}}',
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:788]: Please explain.

How is this related to fixing the vertical alignment?

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 4]
```diff
@@ -104,6 +106,8 @@
 
 			&-text {
 				width: 100%;
+				display: flex;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:109]: If we add the vertical alignment to the `.e-n-tab-text` selector, do we still need it for the `.e-n-tab-icon` as well?

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 5]
```diff
@@ -104,6 +106,8 @@
...
+			  	align-items: center;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:110]: Please use css variables consistently. Why we don't have a css selector here, but we do use it for the icon?

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+			await wpAdmin.setExperiments( {
+				container: false,
+				'nested-elements': false,
+			} );
+		} );
+	}
+} );
+
+test.describe( 'Nested Tabs - Inline font icons experiment', async () => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:108]: Please use one `test.describe()` for all nested tabs tests.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+		// Reset experiments.
+		await wpAdmin.setExperiments( {
+			container: false,
+			'nested-elements': false,
+			e_font_icon_svg: false,
+		} );
+	} );
+
+	test( `Should change icon size`, async ( { page }, testInfo ) => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:151]: For which task it this?

File: tests/playwright/sanity/modules/tabs-v2/tabs-v2.test.js [hunk 8]
```diff
@@ -1,65 +0,0 @@
-const { test, expect } = require( '@playwright/test' );
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/tabs-v2/tabs-v2.test.js:1]: Please sync with latest version of 3.10.0. This test should have been removed in the main repository aleady.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 9]
```diff
@@ -0,0 +1,231 @@
+const { test, expect } = require( '@playwright/test' );
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:1]: There seems to have been a merging issue by Tziv. I am trying to fix that first.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+		} );
+	} );
+} );
+
+test.describe( 'Nested Tabs - Tab icon vertical alignment', async () => {
+	let locationOption;
+	const locationOptions = [ 'Start', 'Center', 'End', 'Justified' ];
+
+	for ( locationOption of locationOptions ) {
+		test( `Should be vertically centered when Justify is set to ${ locationOption }`, async ( { page }, testInfo ) => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:73]: I believe it should always be vertically aligned. I don't think it depends on the location option. Or does it?

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 1 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:149]: If we set `initial` we receive a huge SVG image.
We have to supply a fixed size, to begin with.
At least that is what I saw has been done in other places that use SVG icons

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 2 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:55]: To supply the SVG with an initial fixed size.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 3 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:788]: It isn't that task was updated, and this was added to this task.
fix the SVG icon is not visible in frontend when `inline font icons` experiment is on.

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 4 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:109]: Good comment, I the flex from the icon `.e-n-tab-icon`.
Don't see changes.

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 5 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:110]: In this case, I didn't use variables because there is no specification that this will be changed. 
The text will always be centered.

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 11]
```diff
@@ -124,6 +128,7 @@
 
 			.e-n-tab-icon {
 				display: flex;
+			  	align-items: var( --n-tabs-icon-align-items );
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:131]: Possibly we can delete this and apply this as tab level.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 6 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:108]: We should discuss this.
Not sure that using one `describe` per file is the best practice.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 7 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:151]: The task was updated.
The test purpose is to check if the icon size is changing in the frontend.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 9 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:1]: o.k

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 10 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:73]: No, it doesn't depend on the location.
I am checking that the icon is centered vertically aligned for each of the justify options.
That is why the test's name is like this.

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 11 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:131]: If I apply this to tab level the icon is not vertically centered.

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 12]
```diff
@@ -959,7 +959,7 @@ protected function render() {
 		$this->add_render_attribute( 'elementor-tabs', 'class', 'e-n-tabs' );
 		$this->add_render_attribute( 'tab-title-text', 'class', 'e-n-tab-title-text' );
 		$this->add_render_attribute( 'tab-icon', 'class', 'e-n-tab-icon' );
-		$this->add_render_attribute( 'tab-icon-active', 'class', 'e-active' );
+		$this->add_render_attribute( 'tab-icon-active', 'class', [ 'e-n-tab-icon', 'e-active' ] );
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:962]: Nice work! I had missed that one.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 13]
```diff
@@ -0,0 +1,231 @@
+const { test, expect } = require( '@playwright/test' );
+const WpAdminPage = require( '../../../pages/wp-admin-page' );
+const EditorPage = require( '../../../pages/editor-page' );
+
+test.describe( 'Nested Tabs tests', () => {
+	test( 'Count the number of icons inside the Add Section element', async ( { page }, testInfo ) => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:6]: This is indicated as a new test. Once we compare the right version of 3.10.0 that shouldn't be the case anymore.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+		await expect( editor.getPreviewFrame().locator( '.elementor-widget-n-tabs .e-n-tabs-heading .e-n-tab-title.e-active' ) ).toHaveCSS( 'justify-content', 'flex-start' );
+
+		await wpAdmin.setExperiments( {
+			container: false,
+			'nested-elements': false,
+		} );
+	} );
+} );
+
+test.describe( 'Nested Tabs - Tab icon vertical alignment', async () => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:68]: Please put inside the original `test.describe()`

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+			await wpAdmin.setExperiments( {
+				container: false,
+				'nested-elements': false,
+			} );
+		} );
+	}
+} );
+
+test.describe( 'Nested Tabs - Inline font icons experiment', async () => {
+	test( `Icon should be visible when Inline font icons experiment is active`, async ( { page }, testInfo ) => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:109]: Suggested test name:
`Check visibility of icon svg file when font icons experiment is active`

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+		await editor.publishAndViewPage();
+
+		await page.waitForSelector( '.elementor-widget-n-tabs' );
+
+		const icon = await page.locator( '.elementor-widget-n-tabs .e-n-tab-title .e-n-tab-icon' ).first(),
+			activeTabIcon = await page.locator( '.elementor-widget-n-tabs .e-n-tab-title .e-n-tab-icon.e-active' ).first(),
+			currentContext = page;
+
+		// Assert
+		await expect( activeTabIcon ).toBeVisible();
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:138]: Have you tested this with the original styling?
Did the `.toBeVisible()` fail?

That would be good to test.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+
+		const editor = new EditorPage( page, testInfo ),
+			container = await editor.addElement( { elType: 'container' }, 'document' );
+
+		// Add widgets.
+		await editor.addWidget( 'nested-tabs', container );
+		await editor.getPreviewFrame().waitForSelector( '.e-n-tabs-content .e-con.e-active' );
+
+		// Set icons to tabs according 'TabsIcons' array.
+		await setIconsToTabs( page, TabsIcons );
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:127]: I would remove all the empty lines in between `await` lines.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 13 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:6]: o.k

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 2 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:55]: Typography control fixed.

we can use 1 rem for the icons.
I didn't use the rems course I saw that you used px in all places.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 15 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:109]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 9 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:1]: I think that the tests file name should be `nested-tabs.test.js`

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 18]
```diff
@@ -104,6 +108,8 @@
 
 			&-text {
 				width: 100%;
+				display: var( --n-tabs-title-text-display );
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:111]: I am thinking if we always use center and flex, then we don't need a css variable. I think you can style this without css variables.

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 19]
```diff
@@ -35,6 +35,8 @@
 	--n-tabs-title-background-color-active: var( --e-global-color-accent, var( --n-tabs-color-accent-fallback ) );
 	--n-tabs-title-width: initial;
 	--n-tabs-title-height: initial;
+  	--n-tabs-title-text-display: flex;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:38]: Remove. See below.

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 20]
```diff
@@ -35,6 +35,8 @@
...
+  	--n-tabs-title-text-align-items: center;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:39]: Remove. See below.

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 21]
```diff
@@ -47,10 +49,12 @@
 	--n-tabs-title-border-radius: initial;
 	--n-tabs-title-transition: 0.3s;
 
+  	--n-tabs-icon-align-items: center;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:52]: Remove. See below.

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 22]
```diff
@@ -47,10 +49,12 @@
...
 	--n-tabs-icon-color: var( --e-global-color-secondary, var( --n-tabs-color-secondary-fallback ) );
 	--n-tabs-icon-color-hover: var( --n-tabs-title-color-hover );
 	--n-tabs-icon-color-active: #{$white};
 	--n-tabs-icon-size: initial;
+  	--n-tabs-svg-icon-size: 1rem;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:57]: Change as requested yesterday. Please use my method.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 13 above)
Comment (hein-obox) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:6]: I have sent a fix. That was my error.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 23]
```diff
@@ -0,0 +1,232 @@
...
+
+		await wpAdmin.setExperiments( {
+			container: false,
+			'nested-elements': false,
+		} );
+	} );
+
+	let locationOption;
+
+	const locationOptions = [ 'Start', 'Center', 'End', 'Justified' ];
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:69]: I believe this test is irrelevant. I dont' think the alignment is related to this setting. Can we remove these tests completely?

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 16 above)
Comment (hein-obox) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:137]: This is the most important question for this test. I am not a 100% sure how the `.toBeVisible()` works.

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 1 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss]: I have implemented the changes from your PR.

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 2 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss]: I have implemented the changes from your PR.
This line was deleted.

File: tests/playwright/sanity/modules/tabs-v2/tabs-v2.test.js (hunk 8 above)
Comment (davseve) [reply tests/playwright/sanity/modules/tabs-v2/tabs-v2.test.js]: Done

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 16 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:135]: `ToBeVisible` checks that the element bounding box is greater than zero and that the element is not set to `display hidden`
This test was added because the bug was that SVG icons were not visible.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 3 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php]: Reverted to original code

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 7 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js]: Test title updated by suggestion.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 17 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:125]: Done

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 18 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:110]: Done

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 19 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:38]: Done

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 20 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:39]: Done

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 21 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:53]: Done

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 22 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss]: Changed to your method

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 23 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:69]: Test removed completley

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 24]
```diff
@@ -62,4 +63,127 @@ test.describe( 'Nested Tabs tests', () => {
...
+
+		// Reset experiments.
+		await wpAdmin.setExperiments( {
+			container: false,
+			'nested-elements': false,
+			e_font_icon_svg: false,
+		} );
+	} );
+
+	test( `Check the icon size on frontEnd`, async ( { page }, testInfo ) => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:108]: * on the frontend

**Review by hein-obox — DISMISSED** [review]

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 25]
```diff
@@ -137,12 +140,12 @@
...
 
 					> i {
-						font-size: var( --n-tabs-icon-size );
+					  	font-size: var( --n-tabs-icon-size, var( --n-tabs-title-font-size ) );
 					}
 
-					svg {
-						width: var( --n-tabs-icon-size );
-						height: var( --n-tabs-icon-size );
+				  	svg {
```
Comment (udidol) [inline modules/nested-tabs/assets/scss/frontend.scss:146]: spaces?

**Review by udidol — CHANGES_REQUESTED** [review]

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 25 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss]: Spaced have been removed.

**Review by udidol — APPROVED** [review]
